File: LogViewerControl/VB/Applications/WinFormsLoggingVBNoDI/DataStores/MainControlsDataStore.vb
﻿Imports LogViewerVB.Core
Imports LogDataStore = LogViewerVB.WinForms.Logging.LogDataStore

Namespace DataStores

    ' Application-wide shared instance of the LogDataStore logging entries
    Public Module MainControlsDataStore

        Public Property DataStore As ILogDataStore = New LogDataStore()

    End Module

End Namespace

File: LogViewerControl/CSharp/Applications/WinFormsNLogDI/DataStores/MainControlsDataStore.cs
﻿using LogViewer.Core;
using LogViewer.WinForms;

namespace WinFormsNLogDI.DataStores;

public class MainControlsDataStore
{
    #region Constructors

    public MainControlsDataStore(LogViewerControl logViewer,
                                 ILogDataStore dataStore)
    {
        LogViewer = logViewer;
        DataStore = dataStore;
    }

    #endregion

    #region Properties

    public LogViewerControl LogViewer { get; }

    public ILogDataStore DataStore { get; set; }

    #endregion
}